### PR TITLE
ZCS-2414 Universal UI: Add External Calendar dialog

### PR DESCRIPTION
**Pull request by silentsakky**
- Remove first screen to choose yahoo or other calendar type, we will default to other type only
- UX changes as per mockup
- increase dialog width to 400px

### CODE REVIEW

File: WebRoot/js/zimbraMail/calendar/view/ZmExternalCalendarDialog.js [hunk 1]
```diff
@@ -109,8 +111,8 @@ function(ev) {
                 this.showView(ZmExternalCalendarDialog.SECOND_VIEW, ZmExternalCalendarDialog.TYPE_OTHER);
             }
             this.showIcalView(false);
-            this._syncTypeSelect.setSelectedValue(ZmExternalCalendarDialog.SYNC_TYPE_CALDAV);
-        break;
+            this._syncTypeSelect.setSelectedValue(ZmExternalCalendarDialog.SYNC_TYPE_CALDAV);*/
+        //break;
```
Comment (tarangkhandelwal) [inline WebRoot/js/zimbraMail/calendar/view/ZmExternalCalendarDialog.js]: can we remove this commented line?

File: WebRoot/js/zimbraMail/calendar/view/ZmExternalCalendarDialog.js [hunk 2]
```diff
@@ -149,13 +151,13 @@ ZmExternalCalendarDialog.prototype._backButtonListener =
 function() {
     var id = this._htmlElId;
     switch(this.currentView) {
-        case ZmExternalCalendarDialog.FIRST_VIEW :
-            //this.showView(ZmExternalCalendarDialog.FIRST_VIEW);
+        /*case ZmExternalCalendarDialog.FIRST_VIEW :
```
Comment (tarangkhandelwal) [inline WebRoot/js/zimbraMail/calendar/view/ZmExternalCalendarDialog.js]: can we remove the commented block?

File: WebRoot/js/zimbraMail/calendar/view/ZmExternalCalendarDialog.js (hunk 1 above)
Comment (silentsakky) [reply WebRoot/js/zimbraMail/calendar/view/ZmExternalCalendarDialog.js]: fixed

File: WebRoot/js/zimbraMail/calendar/view/ZmExternalCalendarDialog.js (hunk 2 above)
Comment (silentsakky) [reply WebRoot/js/zimbraMail/calendar/view/ZmExternalCalendarDialog.js]: fixed

File: WebRoot/css/zm.css [hunk 3]
```diff
@@ -7147,6 +7143,28 @@ a.ZmLinkDisabled {
 	@FullWidth@
 }
 
+/* Add External calendar dialog */
```
Comment (kalpeshhpatel) [inline WebRoot/css/zm.css]: Minor: alignment is bit different across different styles.

File: WebRoot/js/zimbraMail/calendar/view/ZmExternalCalendarDialog.js [hunk 4]
```diff
@@ -27,20 +27,22 @@ ZmExternalCalendarDialog = function(params) {
...
     var parent = params.parent || appCtxt.getShell();
     this._controller = params.controller;
-    ZmDialog.call(this, {parent:parent, className:"ZmExternalCalendarDialog", standardButtons:[DwtDialog.NO_BUTTONS], extraButtons: [back, next, cancel], id:'ADD_EXTERNAL_CAL_DIALOG'});
+    ZmDialog.call(this, {parent:parent, className:"ZmExternalCalendarDialog", standardButtons:[DwtDialog.NO_BUTTONS], extraButtons: [back, cancel, next], id:'ADD_EXTERNAL_CAL_DIALOG'});
 
 	this.setButtonListener(ZmExternalCalendarDialog.BACK_BUTTON, new AjxListener(this, this._backButtonListener));
 	this.setButtonListener(ZmExternalCalendarDialog.NEXT_BUTTON, new AjxListener(this, this._nextButtonListener));
 	this.setButtonListener(ZmExternalCalendarDialog.SHARE_CANCEL_BUTTON, new AjxListener(this, this._cancelButtonListener));
 
+	this.getButton(ZmExternalCalendarDialog.NEXT_BUTTON).addClassName('ZPrimaryButton');
```
Comment (kalpeshhpatel) [inline WebRoot/js/zimbraMail/calendar/view/ZmExternalCalendarDialog.js]: Minor: alignment fro line 32 to 36. It was already misaligned. can we correct it?

**Comment by kalpeshhpatel** [comment]
Since, we are replacing first view with second one, can we have same constant ie.`ZmExternalCalendarDialog.FIRST_VIEW`, instead of replacing occurrence of `FIRST_VIEW` with `SECOND_VIEW` for eg. here: https://github.com/Zimbra/zm-web-client/pull/140/files#diff-6751dfbc71ba7f2ce3e80f141fa9f7edR45

And accordingly `SECOND_VIEW` becomes `THIRD_VIEW`.

So our two views will be `FIRST_VIEW` and `SECOND_VIEW`. Currently it's `SECOND_VIEW` and `THIRD_VIEW`, which gives impression of presence of `FIRST_VIEW` which is missing now.

Let me know if you have any question and if this change is bigger.

Thanks

File: WebRoot/js/zimbraMail/calendar/view/ZmExternalCalendarDialog.js (hunk 4 above)
Comment (silentsakky) [reply WebRoot/js/zimbraMail/calendar/view/ZmExternalCalendarDialog.js]: fixed in https://github.com/Zimbra/zm-web-client/pull/140/commits/c476e679237602e6d9249ab741066eb924ddcdd0

File: WebRoot/css/zm.css (hunk 3 above)
Comment (silentsakky) [reply WebRoot/css/zm.css]: Fixed in https://github.com/Zimbra/zm-web-client/pull/140/commits/c476e679237602e6d9249ab741066eb924ddcdd0

**Comment by silentsakky** [comment]
Fixed in https://github.com/Zimbra/zm-web-client/pull/140/commits/c476e679237602e6d9249ab741066eb924ddcdd0

**Pull request merged**